example-async.js

```diff
@@ -0,0 +1,149 @@
+'use strict'
+
+const Fastify = require('fastify')
+
+function build (opts) {
+  const fastify = Fastify(opts)
+
+  fastify
+    .register(require('fastify-jwt'), { secret: 'supersecret' })
+    .register(require('fastify-leveldb'), { name: 'authdb-async' })
+    .register(require('./fastify-auth'))
+    .after(routes)
+
+  fastify.decorate('verifyJWTandLevel', verifyJWTandLevel)
+  fastify.decorate('verifyUserAndPassword', verifyUserAndPassword)
+
+  function verifyJWTandLevel (request, reply) {
+    const jwt = this.jwt
+    const level = this.level
+
+    if (request.body && request.body.failureWithReply) {
+      reply.code(401).send({ error: 'Unauthorized' })
+      return Promise.reject(new Error())
+    }
+
+    if (!request.req.headers['auth']) {
+      return Promise.reject(new Error('Missing token header'))
+    }
+
+    return new Promise(function (resolve, reject) {
+      jwt.verify(request.req.headers['auth'], function (err, decoded) {
+        if (err) { reject(err) };
```

Comment:
We should add a return after the reject otherwise we will cause a memory leak (and we don't want to teach something wrong to users 😁)

Reply:
done, thanks